src/hooks/useNetwork.ts

```diff
@@ -1,16 +1,17 @@
 import {useRef, useContext, useEffect} from 'react';
 import {NetworkContext} from '../components/OnyxProvider';
 
-/**
- * @param {Object} [options]
- * @param {Function} [options.onReconnect]
- * @returns {Object}
- */
-export default function useNetwork({onReconnect = () => {}} = {}) {
+type UseNetworkProps = {
+    onReconnect?: () => void;
+};
+
+type UseNetwork = {isOffline?: boolean};
+
+export default function useNetwork({onReconnect = () => {}}: UseNetworkProps = {}): UseNetwork {
     const callback = useRef(onReconnect);
     callback.current = onReconnect;
 
-    const {isOffline} = useContext(NetworkContext);
+    const {isOffline} = useContext(NetworkContext) ?? {};
```

Comment:
NetworkContext is typed as `React.Context<OnyxKeyValue<"network">>`, and `OnyxKeyValue` currently is nullable. However `NetworkContext` is created with `defaultValue` which is set to `{}` so `NetworkContext` shouldn't ever be nullable (because of the implementation):
```
if (propsToPass[propName] === undefined && defaultValue) {
    propsToPass[propName] = defaultValue;
}
```
Have you tried to make `creteOnyxContext` to return nonnullable Context when provided with `defaultValue`?